linkerd/main/src/main/scala/io/buoyant/linkerd/Main.scala

```diff
@@ -52,6 +57,26 @@ object Main extends App {
     }
   }
 
+  // Just logs the availability of tcnative ssl binding, for troubleshooting
+  private def reportOpenSSLVersion(config: Linker.LinkerConfig): Unit = {
+    if (config.routers.exists(_.servers.exists(_.tls.isDefined))) {
+      log.info(s"TLS required, looking for tcnative binding ...")
+      try {
+        if (OpenSsl.isAvailable) {
+          val version = OpenSsl.versionString()
+          val ciphers = OpenSsl.availableOpenSslCipherSuites()
+          log.info(s"Native OpenSSL available! version: $version, ciphers: $ciphers")
+        } else {
+          log.warning(OpenSsl.unavailabilityCause(), "No native OpenSSL available")
```

Comment:
If I understand correctly we want to configure using FIPS in certain environments such as centos builds on govcloud and we do not care about other environments how we want TLS to behave.

I think rather than letting the library decide based on the availability of certain libraries in the OS which one it uses and logging a warning we might want to have a configurable parameter to explicitly callout the expected behavior. In that case if someone upgrades centos version or introduces incompatible library in the kernel or if the library is missing then we can have a fatal error rather than a warning.

Reply:
As discussed, created PAAS-286. Thanks!